contracts/javascore/xcall/src/main/java/foundation/icon/btp/xcall/CallServiceImpl.java

```diff
@@ -374,34 +390,31 @@ public void onChanCloseConfirm(String portId, String channelId) {
     }
 
     @External
-    public byte[] onRecvPacket(Packet _pct, Address _relayer) {
+    public byte[] onRecvPacket(byte[] calldata, Address relayer) {
         onlyIBCHandler();
-        byte[] _msg = _pct.getData().getBytes();
-        String _from = _pct.getSourcePort() + "/" + _pct.getSourceChannel();
-        BigInteger _sn = _pct.getSequence();
 
-        handleBTPMessage(_from, _from, _sn, _msg);
+        BigInteger newRecvCount = recvCount.getOrDefault(BigInteger.ZERO).add(BigInteger.ONE);
+        recvCount.set(newRecvCount);
 
+        Packet packet = Packet.decode(calldata);
+        byte[] _msg = packet.getData();
+        String _from = packet.getSourcePort() + "/" + packet.getSourceChannel();
+        BigInteger _sn = packet.getSequence();
 
-        return new byte[0];
-
+        handleReceivedPacket(_from, _from, _sn, _msg);
```

Comment:
Ports and channel could be checked here to be correct

Reply:
done